### PR TITLE
[7] Deployments Table - show rollback link

### PR DESCRIPTION
**Pull request by philipnrmn**
This PR:

- [x] adds a deployment link to the deployments table
- [x] adds an action to the marathon store to delete the deployment
- [x] introduces a confirmation modal
- [x] removes the deployment from the store if the delete request is succesful
- [x] handles errors on the delete request

This PR depends on ~~#372~~

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @daiyiwhale

**Comment by aldipower** [comment]
Possibly not related to this PR. Feel free to ignore, should not be a blocker.
----
If I create a fresh service and then click on the deployment tab, I receive this error:
`Cannot read property 'getId' of undefined in DeploymentsTab.js?3396:66`

**Comment by aldipower** [comment]
The Rollback button looks great and triggers the rollback successfully.
Although I miss a little bit UX feedback after I clicked on it. Very silent.

**Comment by aldipower** [comment]
You could consider a confirm dialog before the actual rollback is triggered. 
Like this or in Marathon UI:
https://github.com/dcos/dcos-ui/blob/feature/1066-service-actions/src/js/components/ServiceDetail.js#L111
Just an idea..

**Comment by aldipower** [comment]
Besides the small bug.
It LGTM!

**Comment by philipnrmn** [comment]
Thanks for the extensive feedback @aldipower! Yes, this is a WIP so we're still missing the modal. The issue you're seeing here https://github.com/dcos/dcos-ui/pull/388#issuecomment-225147651 is resolved by #378 and can be ignored here.

**Comment by aldipower** [comment]
OK, very cool!

**Comment by philipnrmn** [comment]
✔️ This PR is ready for review.

**Comment by philipnrmn** [comment]
retest this please

**Comment by philipnrmn** [comment]
screen shot 2016-06-13 at 11 51 22

**Comment by philipnrmn** [comment]
retest this please

**Comment by philipnrmn** [comment]
retest this please

File: src/js/events/__tests__/MarathonActions-test.js [hunk 1]
```diff
@@ -353,4 +353,68 @@ describe('MarathonActions', function () {
...
+
+      it('is made', function () {
+        expect(RequestUtil.json).toHaveBeenCalled();
+      });
+
+      it('is a DELETE', function () {
+        expect(this.configuration.method).toEqual('DELETE');
+      });
+
+      it('calls the approprate endpoint', function () {
```
Comment (orlandohohmeier) [inline src/js/events/__tests__/MarathonActions-test.js]: `appropriate`?

File: src/js/utils/StringUtil.js [hunk 2]
```diff
@@ -66,6 +66,23 @@ const StringUtil = {
     return string.charAt(0).toUpperCase() + string.slice(1, string.length);
   },
 
+  humanizeArray: function (array, serialComma=true) {
```
Comment (orlandohohmeier) [inline src/js/utils/StringUtil.js]: Any chance we can merge this with `arrayToJoinedString`?

**Comment by orlandohohmeier** [comment]
A few minor nits, but this looks pretty good! Great job 💯 ✨

**Comment by orlandohohmeier** [comment]
rollback-issue

The rollback process looks a little strange...

File: src/js/utils/StringUtil.js (hunk 2 above)
Comment (philipnrmn) [inline src/js/utils/StringUtil.js]: Thanks for pointing this out - I didn't realise `arrayToJoinString` existed!

Having poked around the code a bit, it looks like `arrayToJoinedString` is only used to join configuration parameters in the `ConfigurationView`. Do you reckon the intents are similar enough to warrant joining them?

**Comment by philipnrmn** [comment]
Yes, I agree. What do you suggest? Put the deployment into a disabled state until the next poll of the deployments, then replace it?

File: src/js/utils/StringUtil.js (hunk 2 above)
Comment (orlandohohmeier) [inline src/js/utils/StringUtil.js]: I'd say so, in both cases it's used to display arrays to the user. But maybe we just drop `arrayToJoinString` and use your implementation instead. Let's create a tech debt issue to track this and move on with this PR.

**Comment by philipnrmn** [comment]
@orlandohohmeier and I had a chat; we think that the right way to do things is to mark the old deployment as 'rolling back' until the new data comes in to make it clear that one has replaced the other. @leemunroe, what do you reckon?

**Comment by leemunroe** [comment]
👍  Good idea. Much better I think.

**Pull request merged**